Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py

```diff
@@ -16,23 +13,68 @@ class Client(BaseClient):
     Client for CheckPoint RESTful API.
     Args:
           base_url (str): the URL of CheckPoint.
-          sid (str): CheckPoint session ID of the current user session.
+          sid (str): CheckPoint session ID of the current user session [Optional]
           use_ssl (bool): specifies whether to verify the SSL certificate or not.
           use_proxy (bool): specifies if to use Demisto proxy settings.
     """
 
-    def __init__(self, base_url: str, sid: str, use_ssl: bool, use_proxy: bool, **kwargs):
+    def __init__(self, base_url: str, use_ssl: bool, use_proxy: bool, sid: Optional[str] = None, **kwargs):
         super().__init__(base_url, verify=use_ssl, proxy=use_proxy, **kwargs)
-        self.headers = {'Content-Type': 'application/json',
-                        'X-chkp-sid': sid}
         self.verify = use_ssl
+        self.sid = sid if sid != "None" else None
+
+    @property
+    def headers(self):
+        if self.sid is None:  # for logging in, before self.sid is set
+            return {'Content-Type': 'application/json'}
+        return {'Content-Type': 'application/json', 'X-chkp-sid': self.sid}
+
+    def login(self, username: str, password: str, session_timeout: int, domain_arg: str = None) -> CommandResults:
+        """login to a checkpoint admin account using username and password."""
+
+        json_body = {'user': username, 'password': password, 'session-timeout': session_timeout}
+        if domain_arg:
+            json_body['domain'] = domain_arg
+
+        response = self._http_request(method='POST', url_suffix='login', json_data=json_body, headers=self.headers)
+        sid = response.get('sid', '')
+
+        if sid:
+            self.sid = sid
```

Comment:
@DeanArbel should we set it even if it's `None`/`''` etc?

Reply:
Leave it as is. It was an unsuccessful login, so no need to remove an (potential) existing session, that's (could be) stored in `self.sid`.